gems/aws-sdk-core/lib/aws-sdk-core/json/handler.rb

```diff
@@ -9,10 +9,25 @@ class Handler < Seahorse::Client::Handler
       # @param [Seahorse::Client::RequestContext] context
       # @return [Seahorse::Client::Response]
       def call(context)
+        thread = Thread.current
+        t_ser_start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
         build_request(context)
+        t_ser_end = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+        t_deser_start, t_deser_end = nil
+        success = false
         response = @handler.call(context)
-        response.on(200..299) { |resp| parse_response(resp) }
+        response.on(200..299) do |resp|
+          success = true
+          t_deser_start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+          parse_response(resp)
+          t_deser_end = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+        end
         response.on(200..599) { |_resp| apply_request_id(context) }
+        if thread[:warm] && success
+          thread[:json_serde_data] << [format('%.3f', (t_ser_end - t_ser_start) * 1000.0).to_f,
```

Comment:
Is there a reason to combine the serialization and deseralization into the same thread local?  It might simplify to store them separately.